clang/test/Analysis/string.c

```diff
@@ -97,6 +97,17 @@ void strlen_constant2(char x) {
   clang_analyzer_eval(strlen(a) == 3); // expected-warning{{UNKNOWN}}
 }
 
+const char *const global_str_ptr = "abcd";
```

Comment:
What about a non-const global pointer? Could you add a test making sure we do not calculate the wrong size in that case?

Reply:
I've added a new test to cover this too. Since non-const pointers can be changed to point to strings of different length, I've assumed the correct behavior for the comparison is to be UNKNOWN. Please let me know if this is wrong.